### ISSUE TITLE
Mouse click short delay

### ISSUE THREAD

**Issue by artik** [linuxmint/cinnamon#3725]
Strangely, with the new mint 17.1, the mouse need few milliseconds to enable the click. If I'm doing a mouse selection on the desktop for example, the selection start at least 1cm after my click. It's not really noticable if you don't go fast enough.

I tought this bug came from the nvidia drivers, but with my wacom tablet, the click is immediate, as before.

**Comment by zmnpl** [comment]
Just noticed the same. I'll try your workaround. After thinking of your workaround, I assume this might be expected behaviour. As if the system is waiting those ms to see if the second button for middle click emulation will be pressed... In that case it would more likely be a smaller "fail" by design of that routine than a real bug.

**Comment by voycey** [comment]
I am also experiencing this, it only occurs after a suspend but there is at least a 1 second delay between clicking and it actually activating the click.

**Comment by voycey** [comment]
Just and FYI to this as well - It doesn't occur on Mousepad tap to click, only on using the actual button

**Comment by ffflorian** [comment]
I also noticed this and for me @artik's workaround doesn't work :-(

**Comment by megz** [comment]
probably why all the menus are slightly slow on cinnamon, its not the menu per say but the system waiting to finish its process job on the mouse

**Comment by Paviluf** [comment]
The workaround "desactivating the Middle Click emulation" works for me

**Comment by RalfJung** [comment]
I am having the same issue. Selecting something in GIMP by clicking, and then immediately moving the mouse to the other end of the target area, I often ended up with too small a selection. A very confusing experience.

Thanks a lot also for the work-around mentioned here. Disabling the middle-click emulation made everything behave normally again :)

**Comment by g-fresh** [comment]
I'm having the exact same issue as reported by @voycey. Mouse button click delay; touchpad tap-to-click works; happens after hibernation.

Disabling middle mouse button emulation does not fix it.

**Comment by voycey** [comment]
I still get this occasionally - my solution is to hit a terminal and do 

`sudo modprobe -r psmouse && sudo modprobe psmouse`

To reset the mouse subsystem and that usually clears up the problem!

**Comment by Xodetaetl** [comment]
This is still a problem with Cinnamon 3.0.5, click & drag is delayed by a few milliseconds which makes selections and mouse gestures a pain to use as selections do not begin at the position intended and mouse gestures are cut. I have to force myself to wait before starting to drag after clicking.

Disabling middle-click emulation in mouse settings does fix the issue. 
If the delay is necessary for this option to work, then it shouldn't be on by default and there should be a warning in the description.

**Comment by Vahan86** [comment]
@artik, is this still an issue for you in Cinnamon 3.2?

**Comment by Vahan86** [comment]
@JosephMcc, can we close this issue?

**Comment by Drench-97** [comment]
Still an issue in Cinnamon 3.4.4

**Comment by benjamw** [comment]
I'm getting the same issue in 3.8.9+tara.
The workaround does not seem to work.
My click delay is not quite a second, possibly 250-500 ms, but my internal chronometer is not precise enough to make that measurement.

Double clicking is cumbersome, because I have to wait on the first click for the delay, but not so long that I'm past the double click threshold.

**Comment by strongheart1** [comment]
> I still get this occasionally - my solution is to hit a terminal and do
> 
> `sudo modprobe -r psmouse && sudo modprobe psmouse`
> 
> To reset the mouse subsystem and that usually clears up the problem!

I get a delay only on the right mouse button, all other mouse buttons are fine. (Linux Mint 18.3 , Cinnamon 3.6.7 )
The commands [from above](https://github.com/linuxmint/Cinnamon/issues/3725#issuecomment-190951471) seem to fix this for me, too.

**Comment by evranch** [comment]
I was having a similar issue, long delays on right clicks, missing my target on selections, and "dropping" things while dragging them.
I decided to try switching to a different window manager. But when I logged out, I noticed there was an option to start Cinnamon in software rendering mode, so I tried it.

This completely fixed the mouse issue for me.

I'm going to blame ATI and their terrible drivers for this one, like so many other issues I've had since buying the card. Next time it's NVIDIA or bust.

**Comment by icarter09** [comment]
@evranch thanks for the update. I'm going to close this ticket. If anyone is still experiencing this issue in Cinnamon 4.4 (which is the current version at the time of this comment), feel free to open a new ticket and reference this issue. Thanks.

**Comment by christophcemper** [comment]
Just to let you guys know that this is is still an issue in 2020. 

I had to disable that emulation to get mouse clicks respond as fast as usual.
see https://github.com/linuxmint/Cinnamon/issues/3725#issuecomment-65768189

**Comment by MPatzek** [comment]
> I found a workaround, by simply desactivating the Middle Click emulation by click Right+Left button in the mouse settings. I leave open because still a bug for me.

Finally I found this issue. Thank you so much! Unbelievable this is still an issue after almost 7 years.

**Comment by MarjaE2** [comment]
Running into the same problem, but with my trackpad. About half my clicks end up left and/or above where I'd intended.

Disabling right-click and middle-click emulation does not solve the problem, and wouldn't be an solution even if it did.

**Comment by Bratzmeister** [comment]
> > I found a workaround, by simply desactivating the Middle Click emulation by click Right+Left button in the mouse settings. I leave open because still a bug for me.
> 
> Finally I found this issue. Thank you so much! Unbelievable this is still an issue after almost 7 years.

it still is for me 11 years later lol. i enabled middle mouse emulation in dconf editor because in proton-ge my middle mouse click is not working and after weeks of wondering what's going on why fast clicking doesn't lead to shooting how i intend to in games i finally stumbled over this...
